The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4248177851/jobs/7387049933) [(plain)](https://github.com/rust-lang/rust/commit/7ad3a2ba0b65e4f72511ee22639c76625457c13e/checks/11535497319/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:ac593985615ec2ede58e132d2e21d2b1cbd6127c)
Download action repository 'rust-lang/simpleinfra@master' (SHA:697bea7ddceb6696743da8f159f268aef8bfb3c6)
Complete job name: PR (x86_64-gnu-llvm-14, false, ubuntu-20.04-xl)
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: x86_64-gnu-llvm-14
---
   Compiling rustc_session v0.0.0 (/checkout/compiler/rustc_session)
error[E0433]: failed to resolve: use of undeclared crate or module `io`
   --> compiler/rustc_session/src/filesearch.rs:170:54
    |
170 |     fn default_from_current_exe() -> Result<PathBuf, io::Error> {
    |                                                      ^^ use of undeclared crate or module `io`
error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
   --> compiler/rustc_session/src/filesearch.rs:173:29
    |
    |
173 |                 let mut p = canonicalize(exe)?;
    |
    = help: the trait `std::ops::Try` is not implemented for `std::path::PathBuf`

Some errors have detailed explanations: E0277, E0433.
```

</details>

--- a/compiler/rustc_session/src/filesearch.rs
+++ b/compiler/rustc_session/src/filesearch.rs
@@ -182,7 +182,17 @@ pub fn get_or_default_sysroot() -> Result<PathBuf, String> {
         if dir.ends_with(crate::config::host_triple()) {
             dir.parent() // chop off `$target`
                 .and_then(|p| p.parent()) // chop off `rustlib`
-                .and_then(|p| p.parent()) // chop off `lib`
+                .and_then(|p| {
+                    // chop off `lib` (this could be also $arch dir if the host sysroot uses a
+                    // multi-arch layout like Debian or Ubuntu)
+                    match p.parent() {
+                        Some(p) => match p.file_name() {
+                            Some(f) if f == "lib" => p.parent(), // first chop went for $arch, so chop again for `lib`
+                            _ => Some(p),
+                        },
+                        None => None,
+                    }
+                })
                 .map(|s| s.to_owned())
                 .ok_or(format!(
                     "Could not move 3 levels upper using `parent()` on {}",

